src/main/host/thread_ptrace.c

```diff
@@ -659,9 +663,12 @@ void threadptrace_terminate(Thread* base) {
         return;
     }
 
-    if (ptrace(PTRACE_CONT, thread->base.nativePid, 0, SIGTERM) < 0) {
-        warning("ptrace %d: %s", thread->base.nativePid, g_strerror(errno));
+    // need to kill() and not ptrace() since the process may not be stopped
+    if (kill(thread->base.nativePid, SIGKILL) < 0) {
+        warning("kill %d: %s", thread->base.nativePid, g_strerror(errno));
     }
+
+    _threadptrace_nextChildState(thread);
```

Comment:
I'm not sure this is guaranteed to process the "killed" stop; there could be other stops before it. We could call `threadptrace_continue` to process all remaining stops, but perhaps if we get to this point better to just set thread->returnCode right here and return.

Reply:
I'm not sure what you mean here, can't there only be a single stop at a time? If the process is in the stopped state when it receives a sigkill, I would expect a future waitpid() to return the status from the sigkill and not the stop.

Reply:
The child process might have already stopped for another reason before the kill signal was received. I *think* in that case this would see that other stop. The kill signal would still be pending, and would generate another stop after the process is resumed.

Reply:
I think sigkill would be handled differently and wouldn't be queued, since ptrace cannot intercept sigkill (and therefore won't cause a signal-delivery-stop), and the process should exit immediately (regardless of if it's already stopped).

What do you think about the latest commit? I'm fairly confident that the waitpid will return the result from the sigkill, but I added an error() message to check if that's not the case, so hopefully we can tell if this happens in practice while we're still in the testing stages of phantom. I don't think we're any worse off than before.